


The environment I’m using to run sourmash on the HPC is something like this:

	module load GNU/6.2
	module load Python/2.7.2	

and then create & activate a virtualenv and install the following:
	
	virtualenv dev
	source dev/bin/activate
	
(you can put that list in ‘requirements.txt’ and run ‘pip install -r 
requirements.txt’ and it should work.)
	
	mkdir 2017_sourtax_project
	cd 2017_sourtax_project/

upgrade setuptools and pip 

	pip install --upgrade setuptools
	pip install --upgrade pip

make a text file containing install requirements
 
	nano requirements.txt
	
	
and requirements.txt looks like 
	
	bz2file==0.98
	coverage==4.2
	cycler==0.10.0
	matplotlib==1.5.3
	numpy==1.11.2
	py==1.4.31
	pyparsing==2.1.10
	pytest==3.0.4
	pytest-cov==2.4.0
	python-dateutil==2.6.0
	pytz==2016.7
	PyYAML==3.12
	scipy==0.18.1
	screed==0.9
	
and then install requirements
	
	pip install -r requirements.txt
	
Install khmer

	pip install git+https://github.com/dib-lab/khmer.git@master

	
Then finally get the master branch of sourmash 
	
	https://github.com/brooksph/sourmash.git	
	
and install
	
	cd sourmash 
	make install 
			
copy some files over 

	cp /mnt/research/ged/sherine/2015-metagenome-assembly/pipeline/megahit-quality-assembly.fa .

	cp /mnt/research/ged/sherine/2015-metagenome-assembly/pipeline/mircea.fa
	
	
-----------------------------------------------------------------------------

Error trim some seqs 

	trim-low-abund.py -M 16e9 -k 20 SRR606249.pe.qc.fq.gz

-----------------------------------------------------------------------------

Finding unique k-mers 

	unique-kmers.py --report mircea_unique_kmers ../fasta/
	
-----------------------------------------------------------------------------	

#### build signatures from mircea reference genomes from shakya et al 
	sourmash compute -k 21 --dna -n 0 --scaled 10000 mircea.fa
	
#### note: the singleton flag should only be used when the input file consists of a set of genomes not a set of reads

#### make SBT (i.e. database)
	sourmash sbt_index -k 21 --dna mircea ~/2017_sbt_search_project/mircea_project/data/signatures/mircea.fa.sig

#### build a signature from the metagenome (megahit-quality-assembly.fa is the Megahit assembly generated by Sherine)
	sourmash compute -k 21 --dna --scaled 10000 -n 0 megahit-quality-assembly.fa

#### search against mircea
	sourmash sbt_gather -k 21 --dna mircea megahit-quality-assembly.fa.sig threshold=0.001
	
-----------------------------------------------------------------------------

#### compare the ref genome signatures and the read signatures
	sourmash sbt_gather -k 21 --dna mircea ~/2017_sbt_search_project/mircea_project/data/signatures/SRR606249.pe.qc.fq.gz.abundtrim10000.sig threshold=0.0001

##### search missing genomes against ref genomes 

-----------------------------------------------------------------------------

k 21
Compare newly downloaded mircea signatures to an sbt of these signatures and search for missing genomes against this databases 

#### build signatures from mircea reference genomes from shakya et al. Note: when building the database from multiple signatures compute signatures independently and then build the database from them. 

	cat *fna >  mircea_new.fa
	
	sourmash compute -k 21 --dna -n 0 --scaled 10000 mircea_new.fa -o mircea_new.fa.sig
	
	sourmash compute -k 21 --dna -n 0 --scaled 10000 *fna 
	
#### create an sbt to search against 

	sourmash sbt_index -k 21 --dna mircea_new *fna.sig 
	
#### search new sig against sbt 

	sourmash sbt_gather -k 21 mircea_new mircea_new.fa.sig -o mircea_new_sbt_search_k21.txt 
	
k 31 

cat *fna >  mircea_new.fa
	
	sourmash compute -k 31 --dna -n 0 --scaled 10000 mircea_new.fa -o mircea_new.fa.sig -f
	
	sourmash compute -k 31 --dna -n 0 --scaled 10000 *fna -f  
	
#### create an sbt to search against 

	sourmash sbt_index -k 31 --dna mircea_new *fna.sig -f 
	
#### search new sig against sbt 

	sourmash sbt_gather -k 31 mircea_new mircea_new.fa.sig -o mircea_new_sbt_search_k31.txt 
	
k 41

cat *fna >  mircea_new.fa
	
	sourmash compute -k 41 --dna -n 0 --scaled 10000 mircea_new.fa -o mircea_new.fa.sig -f
	
	sourmash compute -k 41 --dna -n 0 --scaled 10000 *fna -f  
	
#### create an sbt to search against 

	sourmash sbt_index -k 41 --dna mircea_new *fna.sig -f 
	
#### search new sig against sbt 

	sourmash sbt_gather -k 41 mircea_new mircea_new.fa.sig -o mircea_new_sbt_search_k41.txt 

k 51

cat *fna >  mircea_new.fa
	
	sourmash compute -k 51 --dna -n 0 --scaled 10000 --singleton mircea_new.fa -o mircea_new.fa.sig -f
	
	sourmash compute -k 51 --dna -n 0 --scaled 10000 *fna -f  
	
#### create an sbt to search against 

	sourmash sbt_index -k 51 --dna mircea_new *fna.sig -f 
	
#### search new sig against sbt 

	sourmash sbt_gather -k 51 mircea_new mircea_new.fa.sig -o mircea_new_sbt_search_k51.txt 


-----------------------------------------------------------------------------



Verifying the mircea ref composition v the microbial database 

	sourmash sbt_gather -k 21 ~ctb/sigs/microbes.sbt.json mircea.fa.sig -o mircea_v_microbes.txt
	
now mircea reads. Calc signature
	
	for i in 500 1000 10000 40000
	do
	
	sourmash compute -k 21 --dna --scaled $i -n 0 SRR606249.pe.qc.fq.gz -o SRR606249.pe.qc.fq.gz.abundtrim${i}.sig
	
	done
	
search against database 

	sourmash sbt_gather -k 21 ~ctb/sigs/microbes.sbt.json SRR606249.pe.qc.fq.gz.abundtrim1000.sig -o mircea-reads-1000_v_microbes.txt

and lindgreen refs

	sourmash sbt_gather -k 21 ~ctb/sigs/microbes.sbt.json lindgreen_ref_seqs.fa -o lindgreen_refs.txt


and run like

qsub test.qsub

which looks like

	#!/bin/sh -login
	#PBS -A ged
	#PBS -l nodes=1:ppn=1
	#PBS -l walltime=72:00:00
	#PBS -l mem=100GB
	#PBS -M philliptbrooks@gmail.com
	
	cd ${PBS_O_WORKDIR}
	source ~/dev/bin/activate
	module load GNU/6.2
	module load Python/2.7.2    
	sourmash sbt_gather -k 21 ~ctb/sigs/microbes.sbt.json SRR606249.pe.qc.fq.gz.abundtrim1000.sig -o mircea-reads-1000_v_microbes.txt
	
	qstat -f $PBS_JOBID

and 

	qsub test.qsub

	
—-----------------------------------------------------------------------------

	sourmash compare -k 21 *interleaved.cat.sub.trim.500.sig



-----------------------------------

Podar data set 

How do we verify that the results that we have are correct? 

We 
Mapping genomes to reference and only use those that 100% match 

Building signatures for lind green signatures using kmer sizes ranging from 17 to 27 	
	sourmash compute -k 17,19,21,23,25,27 --dna -n 0 --scaled 10000 setB2.interleaved.cat.fq.gz


	sourmash compute -k 9,12,15,18,21,24,27,30,33,36 --scaled 10000 --dna -n 0 *interleaved.cat.sub.trim.fq.gz 

Then comparing the signatures 


@PTB create a shell script to automate all of the k sizes

for i in $ 17, 19, 21, 23, 25, 27
do
done

	gunzip -c setA1.interleaved.cat.fq.gz | head -1000000 | trim-low-abund.py -M 16e9 -k 21 - --gzip -o setA1.interleaved.cat.sub.trim.fq.gz
	

thoughts pull out latest version of these genomes with refeseq and then seqech those mathces agaianst the the sbt. It that doesn't work then theres something wrong with method 


-- name may allow you to pull togther a bunch of fasta files 
------------------------------------------------------------------------------

sbt search finds  all matches 

sbt search only finds the best match 

Anything that's in common between the two lists should also be in common between the 

Pull out things that don't match between the ref and the reads. 


----------------------------------------------------------------------------

Lindgreen

----------------------------------------------------------------------------

-----------------------------------------------------------------------------
subsampling reads 

Unzips the file displays the first 400 million reads, puts them in a file setA1.interleaved.cat.sub.fq.gz, and then 

	nano testing_error_trim.sh
	
and testing_error_trim.sh looks like (@PTB modify file to add)
	
	gunzip -c setA1.interleaved.cat.fq.gz | head -10000000 | trim-low-abund.py -M 16e9 -k 21 - --gzip -o setA1.interleaved.cat.sub.trim.fq.gz

	
	gunzip -c setA2.interleaved.cat.fq.gz | head -10000000 | trim-low-abund.py -M 16e9 -k 21 - --gzip -o setA2.interleaved.cat.sub.trim.fq.gz
	
	
	gunzip -c setA3.interleaved.cat.fq.gz | head -10000000 | trim-low-abund.py -M 16e9 -k 21 - --gzip -o setA3.interleaved.cat.sub.trim.fq.gz


	gunzip -c setB1.interleaved.cat.fq.gz | head -10000000 | trim-low-abund.py -M 16e9 -k 21 - --gzip -o setB1.interleaved.cat.sub.trim.fq.gz

	
	gunzip -c setB2.interleaved.cat.fq.gz | head -10000000 | trim-low-abund.py -M 16e9 -k 21 - --gzip -o setB2.interleaved.cat.sub.trim.fq.gz

	
	gunzip -c setB3.interleaved.cat.fq.gz | head -10000000 | trim-low-abund.py -M 16e9 -k 21 - --gzip -o setB3.interleaved.cat.sub.trim.fq.gz
	

------------------------------------------------------------------------------

calculate signatures like 

	nano calc_lindgreen_signatures.sh

	for i in 500 1000 5000 10000 20000 40000
	do
	
	sourmash compute -k 21,31,36,51,101 --scaled $i --dna -n 0 ../fasta/setA1.interleaved.cat.sub.trim.fq.gz -o setA1.interleaved.cat.sub.trim.${i}.sig
		
	done
	
	for i in 500 1000 5000 10000 20000 40000
	do
	
	sourmash compute -k 21,31,36,51,101 --scaled $i --dna -n 0 ../fasta/setA2.interleaved.cat.sub.trim.fq.gz -o setA2.interleaved.cat.sub.trim.${i}.sig
		
	done
	
	for i in 500 1000 5000 10000 20000 40000
	do
	
	sourmash compute -k 21,31,36,51,101 --scaled $i --dna -n 0 ../fasta/setA3.interleaved.cat.sub.trim.fq.gz -o setA3.interleaved.cat.sub.trim.${i}.sig
		
	done
	
	for i in 500 1000 5000 10000 20000 40000
	do
	
	sourmash compute -k 21,31,36,51,101 --scaled $i --dna -n 0 ../fasta/setB1.interleaved.cat.sub.trim.fq.gz -o setB1.interleaved.cat.sub.trim.${i}.sig
		
	done
	
	for i in 500 1000 5000 10000 20000 40000
	do
	
	sourmash compute -k 21,31,36,51,101 --scaled $i --dna -n 0 ../fasta/setB2.interleaved.cat.sub.trim.fq.gz -o setB2.interleaved.cat.sub.trim.${i}.sig
		
	done
	
	for i in 500 1000 5000 10000 20000 40000
	do
	
	sourmash compute -k 21,31,36,51,101 --scaled $i --dna -n 0 ../fasta/setB3.interleaved.cat.sub.trim.fq.gz -o setB3.interleaved.cat.sub.trim.${i}.sig
		
	done
	

------------------------------------------------------------------------------

and the reference seqs 

	for i in 500 1000 5000 10000 20000 40000
	do
		sourmash compute -k 21,31,36,51,101 --scaled $i --dna -n 0 --singleton ../fasta/ref_sequences/lindgreen_ref_seqs.fa -o lindgreen_ref_seqs${i}.sig
	done
	

----------------------------------------------------------------------------


Questions: 

1. Why doesn't sbt_index accept multiple k values? 




